daycare/test_daycare.py

```diff
@@ -0,0 +1,119 @@
+    return DayCare.create(email=EMAIL, username=USERNAME, password=PASSWORD,
+                          name=NAME, description=DESCRIPTION,
+                          price_per_day=PRICE_PER_DAY,
+                          capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+
+
+@pytest.mark.django_db()
+class TestDaycareModel:
+    def test_persist_daycare(self, create_daycare_user):
+        assert create_daycare_user.user in User.objects.all()
+        assert create_daycare_user in DayCare.objects.all()
+
+    def test_del_daycare(self, create_daycare_user):
+        assert create_daycare_user.user in User.objects.all()
+        assert create_daycare_user in DayCare.objects.all()
+
+        create_daycare_user.delete()
+        assert create_daycare_user not in DayCare.objects.all()
+        assert create_daycare_user not in User.objects.all()
+
+    def test_daycare_user_delete(self, create_daycare_user):
+        assert create_daycare_user.user in User.objects.all()
+        assert create_daycare_user in DayCare.objects.all()
+
+        create_daycare_user.user.delete()
+        assert create_daycare_user not in DayCare.objects.all()
+
+    def test_daycare_user_creation_with_invalid_email(self):
+        with pytest.raises(ValidationError, match="'Enter a valid email address.'"):
+            DayCare.create(username='testuser02', email="invalid mail format", password=PASSWORD,
+                           name=NAME, description=DESCRIPTION,
+                           price_per_day=PRICE_PER_DAY,
+                           capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+
+    def test_daycare_user_creation_with_same_user_name(self):
+        with pytest.raises(ValidationError, match="Invalid username"):
+            DayCare.create(email=EMAIL, username="testuser01", password=PASSWORD,
+                           name=NAME, description=DESCRIPTION,
+                           price_per_day=PRICE_PER_DAY,
+                           capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+            DayCare.create(email=EMAIL, username="testuser01", password=PASSWORD,
+                           name=NAME, description=DESCRIPTION,
+                           price_per_day=PRICE_PER_DAY,
+                           capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+
+    def test_daycare_user_with_invalid_name(self):
+        with pytest.raises(ValidationError, match="Invalid daycare name length - the max length of the name is 20"):
+            DayCare.create(username=USERNAME, email=EMAIL, password=PASSWORD,
+                           name="this is very long name above 20 letters and maybe moreeeeeee",
+                           description=DESCRIPTION,
+                           price_per_day=PRICE_PER_DAY,
+                           capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+            raise ValidationError("Invalid daycare name length - the max length of the name is 20")
```

Comment:
No need to raise this exception, it is already done by `pytest.rasies` call.

Reply:
Done @ormergi